graphql-java-servlet/src/main/java/graphql/kickstart/servlet/HttpRequestHandlerImpl.java

```diff
@@ -49,11 +52,31 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
   private void execute(GraphQLInvocationInput invocationInput, HttpServletRequest request,
       HttpServletResponse response) {
     try {
+      if (configuration.getResponseCache() != null) {
```

Comment:
Bit too much logic here imo. Would be better if you'd move this part into a separate class and just call its method here. If you let it return a `boolean` then you can get rid of that `return;` at line 67 too and it'll be a bit easier to follow.

Reply:
Done.